modules/aws/volume-access/main.tf

```
@@ -0,0 +1,109 @@
+data "aws_caller_identity" "current" {}
+locals {
+  external_id      = (var.external_id != "" ? [{ test : "StringEquals", variable : "sts:ExternalId", values : [var.external_id] }] : [])
+  account_ids      = distinct(concat(var.account_ids, local.default_account_ids))
+  bucket_list      = distinct([for item in var.buckets : "arn:aws:s3:::${split("/", item)[0]}"])
+  bucket_path_list = distinct([for item in var.buckets : "arn:aws:s3:::${item}"])
+  tag_set          = merge({ Vendor = "StreamNative", Module = "StreamNative Volume", SNVersion = var.sn_policy_version }, var.tags)
+  default_account_ids = compact([
+    # will add it in the next pr
+  ])
+  conditions = [
+    for value in local.account_ids :
+    [
+      {
+        test : "StringEquals",
+        variable : "sts:ExternalId",
+        values : [var.external_id]
+      }
+    ]
+  ]
+}
+
+data "aws_iam_policy_document" "streamnative_management_access" {
+  statement {
+    sid     = "AllowStreamNativeControlPlaneAccess"
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = var.streamnative_vendor_access_role_arns
+    }
+    dynamic "condition" {
+      for_each = local.external_id
+      content {
+        test     = condition.value["test"]
+        values   = condition.value["values"]
+        variable = condition.value["variable"]
+      }
+    }
+  }
+
+  dynamic "statement" {
```

This block is duplicated, since the `local.conditions` already composed by the account_ids.

```suggestion
  statement {
```

---

Fixed